### PR TITLE
Fix : issue double fetching utilities

### PR DESCRIPTION
**Pull request by marsellavaleria19**
## Changes
[fix : ssue double fetch utilities](https://github.com/jabardigitalservice/super-app-cms/commit/2cd6f621fd4f353e0ca381d0086a693593adea64)
[fix : some issue on form add complaint](https://github.com/jabardigitalservice/super-app-cms/commit/08e6b806947fdb6d71c1eccfe4535eae269133bc)
[feat : add state management for utilities](https://github.com/jabardigitalservice/super-app-cms/commit/640b69675f7faa250170e99d6ff80eb8c23bf083)

## Evidence
title: fix issue double fetching utilities
project: Jabar Super Apps
participants: @marsellavaleria19 @yoslie @rayfajars @rizfirman @Ibwedagama 
screenshot: https://s.digitalservice.id/fBoSvm

### CODE REVIEW

File: components/Aduan/Dialog/AddComplaint/Form/OthersComplaint/index.vue [hunk 1]
```diff
@@ -111,7 +120,7 @@
 
 <script>
 import { ValidationProvider, ValidationObserver } from 'vee-validate'
-import { ENDPOINT_ADUAN } from '~/constant/endpoint-api'
+// import { ENDPOINT_ADUAN } from '~/constant/endpoint-api'
```
Comment (Ibwedagama) [inline components/Aduan/Dialog/AddComplaint/Form/OthersComplaint/index.vue:123]: hello @marsellavaleria19 is this commented code used?

File: components/Aduan/Dialog/AddComplaint/index.vue [hunk 2]
```diff
@@ -10,16 +10,62 @@
...
+      :is-success="!isError"
+      @retry="backToForm()"
+      @close-all-modal="closePopupInformation()"
+    />
+    <!-- <DialogConfirmation
+      :show-popup="isShowPopupConfirmation"
+      :data-dialog="dataDialog"
+      @submit="submitPopupConfirmationHandle()"
+      @close="closePopupConfirmationHandle()"
+    /> -->
```
Comment (Ibwedagama) [inline components/Aduan/Dialog/AddComplaint/index.vue:60]: this also

File: components/Aduan/Dialog/AddComplaint/index.vue [hunk 3]
```diff
@@ -28,25 +74,33 @@ import { mapGetters } from 'vuex'
 import FormInformationComplaint from '~/components/Aduan/Dialog/AddComplaint/Form/InformationComplaint'
 import FormLocationComplaint from '~/components/Aduan/Dialog/AddComplaint/Form/LocationComplaint'
 import FormOtherComplaint from '~/components/Aduan/Dialog/AddComplaint/Form/OthersComplaint'
-import popupAduanMasuk from '~/mixins/popup-aduan-masuk'
+import { iconPopup } from '~/constant/icon-popup-new'
+// import popupAduanMasuk from '~/mixins/popup-aduan-masuk'
```
Comment (Ibwedagama) [inline components/Aduan/Dialog/AddComplaint/index.vue:78]: please remove the commented code if it's not being used 👍🏼

File: components/Aduan/Dialog/AddComplaint/index.vue [hunk 4]
```diff
@@ -10,16 +10,62 @@
...
+            ref="formInformationComplaint"
+          />
+          <FormLocationComplaint
+            v-show="indexCurrentActive === 2"
+            ref="formLocationComplaint"
+          />
+          <FormOtherComplaint
+            v-show="indexCurrentActive === 3"
+            ref="formOtherComplaint"
+          />
```
Comment (rayfajars) [inline components/Aduan/Dialog/AddComplaint/index.vue:25]: teh ini gak jadi diganti v-if aja ?

kemarin kena issue fetch double

File: components/Aduan/Dialog/ProcessComplaint/index.vue [hunk 5]
```diff
@@ -341,49 +341,53 @@ export default {
...
+  //     const responseGovResponsible = await this.$axios.get(
+  //       `${ENDPOINT_ADUAN}/opds`
+  //     )
+  //     this.listDataGovResponsible = responseGovResponsible.data.data
+  //   } catch {
+  //     this.listDataComplaintStatus = []
+  //     this.listDataAuthority = []
+  //     this.listDataDisposition = []
+  //   }
+  // },
```
Comment (rayfajars) [inline components/Aduan/Dialog/ProcessComplaint/index.vue:371]: is used ?

kalo masih dipake bisa tambahin commend TODO teh @marsellavaleria19

File: components/Aduan/Dialog/ProcessComplaint/index.vue [hunk 6]
```diff
@@ -405,12 +409,27 @@ export default {
...
   watch: {
-    payload() {
-      if (this.payload.coverage_of_affairs) {
-        this.$fetch()
-      }
+    // payload() {
+    //   if (this.payload.coverage_of_affairs) {
+    //     this.$fetch()
+    //   }
+    // },
```
Comment (rayfajars) [inline components/Aduan/Dialog/ProcessComplaint/index.vue:416]: this to

File: components/Aduan/index.vue [hunk 7]
```diff
@@ -850,25 +857,25 @@ export default {
...
+    //       {
+    //         id: '',
+    //         name: 'Semua Kategori Aduan',
+    //       },
+    //       ...this.listDataCategory,
+    //     ]
+    //   } catch (error) {
+    //     console.error(error)
+    //   }
+    // },
```
Comment (rayfajars) [inline components/Aduan/index.vue:878]: this to

File: mixins/popup-aduan-masuk.js [hunk 8]
```diff
@@ -246,6 +252,37 @@ export default {
...
+      // this.setDataDialog({
+      //   dataComplaint,
+      //   ...this.setDataDialogConfirmation(
+      //     'Buat Instruksi',
+      //     'No.Aduan',
+      //     dataComplaint.complaint_id,
+      //     'Buat Instruksi'
+      //   ),
+      //   proposed_ikp_narrative: dataComplaint.proposed_ikp_narrative || '-',
+      // })
```
Comment (rayfajars) [inline mixins/popup-aduan-masuk.js:279]: this to

File: store/utilities-complaint.js [hunk 9]
```diff
@@ -0,0 +1,153 @@
...
+  //     const responseGovResponsible = await this.$axios.get(
+  //       `${ENDPOINT_ADUAN}/opds`
+  //     )
+  //     this.listDataGovResponsible = responseGovResponsible.data.data
+  //   } catch {
+  //     this.listDataDisposition = []
+  //     this.listDataAuthority = []
+  //     this.listDataGovResponsible = []
+  //   }
+  // },
```
Comment (rayfajars) [inline store/utilities-complaint.js:125]: this to

File: components/Aduan/Dialog/AddComplaint/index.vue [hunk 10]
```diff
@@ -100,69 +154,106 @@ export default {
...
+    showPopupConfirmationAddComplaint() {
       if (this.typeConfirmation === 'submit') {
-        this.setDataDialog({
+        this.dialogConfirmation = {
           title: 'Konfirmasi Tambah Aduan',
-          description: 'Apakah Anda yakin ingin menyimpan data ini?',
-          labelButtonSubmit: 'Ya Simpan'
-        })
+          nameModal: `${this.nameModal}Confirmation`,
+          descriptionText: 'Apakah anda yakin ingin menyimpan data ini?',
```
Comment (yoslie) [inline components/Aduan/Dialog/AddComplaint/index.vue:162]: sepertinya yang awal sudah benar penulisan `Anda` dengan a huruf besar *cmiiw

**Review by yoslie — APPROVED** [review]

**Review by rayfajars — APPROVED** [review]

**Review by rizfirman — APPROVED** [review]

File: components/Aduan/Dialog/AddComplaint/Form/OthersComplaint/index.vue (hunk 1 above)
Comment (marsellavaleria19) [reply components/Aduan/Dialog/AddComplaint/Form/OthersComplaint/index.vue]: Hello @Ibwedagama , it's not used. Sorry I'm forget to delete this. I've fix it, this is my commit : https://github.com/jabardigitalservice/super-app-cms/pull/360/commits/8a25256d471ddc875aaf1da575cd9e1487c8dce6

File: components/Aduan/Dialog/AddComplaint/index.vue (hunk 2 above)
Comment (marsellavaleria19) [reply components/Aduan/Dialog/AddComplaint/index.vue]: I've fix it, this is my commit : https://github.com/jabardigitalservice/super-app-cms/pull/360/commits/8a25256d471ddc875aaf1da575cd9e1487c8dce6

File: components/Aduan/Dialog/ProcessComplaint/index.vue (hunk 5 above)
Comment (marsellavaleria19) [reply components/Aduan/Dialog/ProcessComplaint/index.vue]: No. I've fix it, this is my commit : https://github.com/jabardigitalservice/super-app-cms/pull/360/commits/8a25256d471ddc875aaf1da575cd9e1487c8dce6

File: components/Aduan/Dialog/ProcessComplaint/index.vue (hunk 6 above)
Comment (marsellavaleria19) [reply components/Aduan/Dialog/ProcessComplaint/index.vue]: I've fix it, this is my commit : https://github.com/jabardigitalservice/super-app-cms/pull/360/commits/8a25256d471ddc875aaf1da575cd9e1487c8dce6

File: components/Aduan/index.vue (hunk 7 above)
Comment (marsellavaleria19) [reply components/Aduan/index.vue]: I've fix it, this is my commit : https://github.com/jabardigitalservice/super-app-cms/pull/360/commits/8a25256d471ddc875aaf1da575cd9e1487c8dce6

File: mixins/popup-aduan-masuk.js (hunk 8 above)
Comment (marsellavaleria19) [reply mixins/popup-aduan-masuk.js]: I've fix it, this is my commit : https://github.com/jabardigitalservice/super-app-cms/pull/360/commits/8a25256d471ddc875aaf1da575cd9e1487c8dce6

File: store/utilities-complaint.js (hunk 9 above)
Comment (marsellavaleria19) [reply store/utilities-complaint.js]: I've fix it, this is my commit : https://github.com/jabardigitalservice/super-app-cms/pull/360/commits/8a25256d471ddc875aaf1da575cd9e1487c8dce6

File: components/Aduan/Dialog/AddComplaint/index.vue (hunk 4 above)
Comment (marsellavaleria19) [reply components/Aduan/Dialog/AddComplaint/index.vue]: Ah iya lupa aku ganti pake v-if. Udah aku benerin,kang https://github.com/jabardigitalservice/super-app-cms/pull/360/commits/1a5d2422986336e94b4a5436c25acd0251658337

**Pull request merged**